lib/internal/util/comparisons.js

```diff
@@ -82,6 +96,24 @@ function isObjectOrArrayTag(tag) {
   return tag === '[object Array]' || tag === '[object Object]';
 }
 
+function checkBoxedPrimitives(val1, val2) {
```

Comment:
Can you give this a name that includes `deepEqual`? `check` is a bit too generic, imo…

Reply:
I used `isEqualBoxedPrimitive`. I guess that's better?